### PR TITLE
docs: fix missing possibleValues

### PR DESCRIPTION
**Pull request by ckerr**
#### Description of Change

Inspired by https://github.com/electron/electron/pull/19448. Tweaks the docs so that proper `possibleValues` will be generated for:

 * contents.insertCSS()'s `options` parameter's `cssOrigin` property
 * image.resize()'s `options` parameter's `quality` property
 * new ClientRequest()'s `options` parameter's `protocol` property
 * new ClientRequest()'s `options` parameter's `redirect` property

CC @MarshallOfSound @codebytere 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: no-notes

### CODE REVIEW

**Comment by codebytere** [comment]
@ckerr:
```
new ClientRequest()'s options parameter's protocol property
new ClientRequest()'s options parameter's redirect property
```
still aren't showing up in the ts defs file it seems!

File: filenames.auto.gni [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # THIS FILE IS AUTO-GENERATED, PLEASE DO NOT EDIT BY HAND
 auto_filenames = {
   api_docs = [
+    "docs/api/.menu-item.md.swp",
```
Comment (codebytere) [inline filenames.auto.gni]: 🤔 not sure about this extension, was it added on purpose?

File: docs/api/structures/client-request-constructor-options.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# ClientRequestConstructorOptions Object
```
Comment (MarshallOfSound) [inline docs/api/structures/client-request-constructor-options.md]: We probably don't want to do this? This constructor object should be automatically generated by the docs parser, if it's not then that's a bug in the docs parser we should fix

Comment (ckerr) [reply docs/api/structures/client-request-constructor-options.md]: Agreed. I think this is resolved in docs-parser 0.4.2 in that when there's a union (here, an object | string) followed by a key list, docs-parser now only applies the list to the first union type.

**Comment by ckerr** [comment]
Some of this PR overlaps with the changes in https://github.com/electron/electron/pull/19585, and since both PRs rely on docs-parser 0.4.2, this should be updated _after_ 19585 lands

**Comment by codebytere** [comment]
@ckerr the dependent PR is merged, so this should be good to rebase and merge 🚀

**Comment by zcbenz** [comment]
@ckerr Can you rebase this branch?

**Comment by zcbenz** [comment]
Closing since there has been no activity for a long time.

**Pull request closed without merging**